### ISSUE TITLE
feat: add `egctl install` and `egctl uninstall`

### ISSUE THREAD

**Issue by ShyunnY** [envoyproxy/gateway#2711]
*Description*:

Have we considered adding more installation methods for **eg**?

Currently we support installation through `k8s resource file` and `helm`. 

In my opinion, we can add `egctl install` command line method for installation and `egctl uninstall` method for uninstallation.

This can reduce other issues that users need to pay attention to, and reduces the installation steps compared to the other two methods. This is more friendly to users who are less familiar with k8s/helm.

**Comment by ShyunnY** [comment]
An additional question, currently egctl does not support obtaining sds configuration. What is the reason for this?

**Comment by ShyunnY** [comment]
@arkodg

**Comment by Xunzhuo** [comment]
This is a reasonable request and good for UX. Feel free to try to implement it if you are interested in it.

**Comment by ShyunnY** [comment]
> This is a reasonable request and good for UX. Feel free to try to implement it if you are interested in it.

For grabbing sds configuration, do we consider implementing it?

**Comment by ShyunnY** [comment]
Please assign it to me, thanks!

**Comment by ShyunnY** [comment]
When implementing `egctl install`, I utilized the Helm SDK. In the installation of `eg` related resources, I employed two installation strategies:

**Strategy 1:**
+ Approach:   Initially, install CRDs without including them as part of the Helm Release, and then install a Helm Release that does not contain CRDs.
+ Advantage:   This ensures that CRDs will not be deleted in subsequent Helm uninstalls or future `egctl uninstall`. We intend for the deletion of CRDs to be performed only by administrators who understand the consequences of removing CRDs, as this action will result in the garbage collection of all related CRs in the cluster.

**Strategy 2:**
+ Approach:   Install the entire Helm Release without splitting CRDs.
+ Advantage:   During uninstallation, we can perform resource cleanup more quickly without having to consider lingering issues related to CRDs.

Personally, I lean towards casting my vote for Strategy 1.

@arkodg @Xunzhuo @zirain @shawnh2
